java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -116,12 +119,13 @@ public void reportIssue(JavaCheck javaCheck, Tree tree, String message) {
   @Override
   public void reportIssue(JavaCheck javaCheck, Tree syntaxNode, String message, List<Location> secondary, @Nullable Integer cost) {
     List<List<Location>> flows = secondary.stream().map(Collections::singletonList).collect(Collectors.toList());
-    reportIssueWithFlow(javaCheck, syntaxNode, message, flows, cost);
+    sonarComponents.reportIssue(createAnalyzerMessage(file, javaCheck, syntaxNode, null, message, flows, cost));
   }
 
   @Override
   public void reportIssueWithFlow(JavaCheck javaCheck, Tree syntaxNode, String message, Iterable<List<Location>> flows, @Nullable Integer cost) {
-    sonarComponents.reportIssue(createAnalyzerMessage(file, javaCheck, syntaxNode, null, message, flows, cost));
+    // FIXME SONARJAVA-2111 report all flows
```

Comment:
As discussed :  this "fixme" comment and its consequences on the flows list should be on the SE engine side in order to avoid breaking change in the API behavior.